py_ecc/bls/hash.py

```diff
@@ -30,8 +36,31 @@ def hkdf_expand(prk: Union[bytes, bytearray], info: Union[bytes, bytearray], len
         text = previous + info + bytes([i + 1])
 
         # T(i + 1) = HMAC(T(i) || info || i)
-        previous = bytearray(hmac.new(prk, text, hashlib.sha256).digest())
+        previous = bytearray(hmac.new(prk, text, _sha256).digest())
         okm.extend(previous)
 
     # Return first `length` bytes.
     return okm[:length]
+
+
+def sha256(x: bytes) -> bytes:
+    m = _sha256()
+    m.update(x)
+    return m.digest()
+
+
+def xor(a: bytes, b: bytes) -> bytes:
+    return bytes(_a ^ _b for _a, _b in zip(a, b))
+
+
+def expand_message_xmd(msg: bytes, DST: bytes, len_in_bytes: int) -> bytes:
```

Comment:
What do you think about following the description of the input parameters of [the spec](https://tools.ietf.org/html/draft-irtf-cfrg-hash-to-curve-06#section-5.3.1) by abstracting the hash function as one of the properties of the `BaseG2Ciphersuite` implementations?

e.g.,
```python
class G2Basic(BaseG2Ciphersuite):
    DST = b'BLS_SIG_BLS12381G2_XMD:SHA-256_SSWU_RO_NUL_'
    xmd_hash_function = sha256
```